hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -3034,10 +3042,14 @@ private void setOptionalObjectMetadata(ObjectMetadata metadata) {
    * </ol>
    * @param key key written to
    * @param length  total length of file written
+   * @throws IOException if metadata about the write could
```

Comment:
change @throws from IOException to more specific MetadataPersistenceException

Reply:
updated